pkg/registry/apis/provisioning/repository/github/factory.go

```diff
@@ -20,16 +22,15 @@ func ProvideFactory() *Factory {
 	return &Factory{}
 }
 
-func (r *Factory) New(ctx context.Context, ghToken string) Client {
+func (r *Factory) New(ctx context.Context, ghToken common.RawSecureValue) Client {
 	if r.Client != nil {
 		return NewClient(github.NewClient(r.Client))
 	}
 
-	tokenSrc := oauth2.StaticTokenSource(
-		&oauth2.Token{AccessToken: ghToken},
-	)
-
-	if len(ghToken) > 0 {
+	if !ghToken.IsZero() {
+		tokenSrc := oauth2.StaticTokenSource(
+			&oauth2.Token{AccessToken: string(ghToken)},
```

Comment:
Don't you want to do this instead? I guess there's nothing to prohibit anyone from casting it unless we made `RawSecureValue` a struct.
```suggestion
			&oauth2.Token{AccessToken: ghToken.DangerouslyExposeAndConsumeValue()},
```

Reply:
I am nervous about that because the token is used twice:
1. for the nanogit client (does all the common things)
2. for the github client (offers extended features)

the dangerously one enforces that it value is only used once